### PR TITLE
JS: Add data-flow from `yield` in generators.

### PR DESCRIPTION
**Pull request by erik-krogh**
This is a followup to the support I added [for async functions](https://github.com/github/codeql/pull/4019).

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/Functions.qll [hunk 1]
```diff
@@ -204,6 +204,9 @@ class Function extends @function, Parameterized, TypeParameterized, StmtContaine
   /** Holds if this function is an asynchronous function. */
   predicate isAsync() { isAsync(this) }
 
+  /** Holds if this function is not asynchronous and also not a generator. */
+  predicate isOrdinary() { not isAsync() and not isGenerator() }
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/Functions.qll:208]: I'd say there are too many kinds of `Function` instances to have a predicate named `isOrdinary`. Implicit class constructors, abstract methods, bodiless functions in ambient context, function signatures etc, all seem non-ordinary in their own way.

I'd suggest `isAsyncOrGenerator` and then replacing the calls with `not f.isAsyncOrGenerator()`

File: javascript/ql/src/semmle/javascript/dataflow/internal/FlowSteps.qll [hunk 2]
```diff
@@ -71,7 +71,7 @@ predicate localExceptionStep(DataFlow::Node pred, DataFlow::Node succ) {
  */
 predicate localExceptionStepWithAsyncFlag(DataFlow::Node pred, DataFlow::Node succ, boolean async) {
   exists(DataFlow::Node target | target = getThrowTarget(pred) |
-    async = false and
+    async = false and // this also covers generators - as the behavior of exceptions is close enough to the behavior of ordinary functions when it comes to exceptions (assuming that the iterator does not cross function boundaries).
```
Comment (asgerf) [inline javascript/ql/src/semmle/javascript/dataflow/internal/FlowSteps.qll:74]: Break the comment into multiple lines

**Review by asgerf — COMMENTED** [review]
Looking good! Could you run a quick evaluation?

**Comment by erik-krogh** [comment]
> Looking good! Could you run a quick evaluation?

[Evaluation looks fine](https://git.semmle.com/erik/dist-compare-reports/tree/profiling-erik-krogh.northeurope.cloudapp.azure.com_1598985653201)

**Review by asgerf — APPROVED** [review]

**Pull request merged**